lib/spack/spack/cmd/test.py

```diff
@@ -358,18 +358,17 @@ def _report_suite_results(test_suite, args, constraints):
 
         tty.msg("test specs:")
 
-        failed, skipped, untested = 0, 0, 0
+        counts = Counter()
         for pkg_id in test_specs:
             if pkg_id in results:
                 status = results[pkg_id]
-                if status == "FAILED":
-                    failed += 1
-                elif status == "NO-TESTS":
-                    untested += 1
-                elif status == "SKIPPED":
-                    skipped += 1
-
-                if args.failed and status != "FAILED":
+                # Backward-compatibility:  NO-TESTS => NO_TESTS
+                status = "NO_TESTS" if status == "NO-TESTS" else status
+
+                status = spack.install_test.TestStatus[status]
+                counts[status] += 1
```

Comment:
I am using `Counter` to 1) track occurrences of the test status values *and* 2) for the `total()` (see the `parts()` and `write_test_summary()` methods).

Reply:
Sure, the point I'm making is that:
1. For tracking occurences, we're doing it manually (so there's no benefit in using `Counter` which does it in `__init__`)
2. For `Counter.total()` we have the implementation that extract the same value from a generic dict, since the method has been added in Python 3.10

But, as I said, this is very minor, so I'm going to collapse this item.